astronomy/orbital_elements_body.hpp

```diff
@@ -42,17 +44,22 @@ StatusOr<OrbitalElements> OrbitalElements::ForTrajectory(
   orbital_elements.osculating_equinoctial_elements_ =
       OsculatingEquinoctialElements(trajectory, primary, secondary);
   orbital_elements.radial_distances_ = RadialDistances(trajectory);
-  orbital_elements.sidereal_period_ =
+  auto const sidereal_period =
       SiderealPeriod(orbital_elements.osculating_equinoctial_elements_);
+  RETURN_IF_ERROR(sidereal_period.status());
+  orbital_elements.sidereal_period_ = sidereal_period.ValueOrDie();
   if (!IsFinite(orbital_elements.sidereal_period_)) {
     // Guard against NaN sidereal periods (from hyperbolic orbits).
     return Status(
         Error::OUT_OF_RANGE,
         "sidereal period is " + DebugString(orbital_elements.sidereal_period_));
   }
-  orbital_elements.mean_equinoctial_elements_ =
+  auto mean_equinoctial_elements =
       MeanEquinoctialElements(orbital_elements.osculating_equinoctial_elements_,
                               orbital_elements.sidereal_period_);
+  RETURN_IF_ERROR(mean_equinoctial_elements.status());
```

Comment:
We should change the macro to automatically extract the status when give a `base::StatusOr`.